### PR TITLE
908 - Adding configs/secrets to service inspect pretty

### PR DESCRIPTION
**Pull request by essamhassan**
Signed-off-by: Essam A. Hassan <es.hassan187@gmail.com>

**- What I did**
Resolving #908 showing info about configs/secrets in docker service inspect --pretty version

**- How I did it**
1- methods to get configs and secrets
2- adding configs/secrets to the template if they exists

**- How to verify it**

 `docker config create configtest.conf`

`docker secret create secrettest.conf`

```bash
    docker service create -d \
    --config source=foo.conf,target=/foo.conf,uid=123,gid=456 \
    --secret source=secret.conf,target=/secret.conf,uid=234,gid=567 \
    --replicas 1 \
    --name myservice \
    nginx:alpine
```
Output should be
```bash
ID:		v6hlue5d5g0yhethbsxwsuvgp #other id
Name:		myservice
Configs:
 Target:	/foo.conf
  Source:	foo.conf
Secrets:
 Target:	/secret.conf
  Source:	secret.conf
Service Mode:	Replicated
 Replicas:	1
Placement:
UpdateConfig:
 Parallelism:	1
 On failure:	pause
 Monitoring Period: 5s
 Max failure ratio: 0
 Update order:      stop-first
RollbackConfig:
 Parallelism:	1
 On failure:	pause
 Monitoring Period: 5s
 Max failure ratio: 0
 Rollback order:    stop-first
ContainerSpec:
 Image:		nginx:alpine@sha256:89f15d9cbad18d54fe49acc79b24199a05b4b1c48294ce4516393bdd2bc714b2
Resources:
Endpoint Mode:	vip

```
**- Description for the changelog**

adding configs/secrets to service inspect pretty
**- A picture of a cute animal (not mandatory but encouraged)**

Very important cat!
A smart cat

### CODE REVIEW

**Comment by thaJeztah** [comment]
I also noticed that `Mounts` is using `=` between field and value (but all other parts use a colon (`:`); we should change that (but in a separate PR)

**Comment by essamhassan** [comment]
thank you for reviewing the pr

- concerning the amount of displayed data, I actually agree that --pretty should match the json output however I noticed that there are so many trimmed data from the json output that's why I decided to go for the simpler output. If we agreed on changing this, I can push the changes and update the branch

- For the mounts, its a simple change I guess I can open another pr for it

- I also want to note that I think the testing for service inspect does not cover all cases, my proposal is build a factory for service at `formatServiceInspect` so that future tests can be flexible and not work on a static well defined state of the service

File: cli/command/service/inspect_test.go [hunk 1]
```diff
@@ -133,3 +145,14 @@ func TestJSONFormatWithNoUpdateConfig(t *testing.T) {
 	}
 	assert.Check(t, is.DeepEqual(m1, m2))
 }
+
+func TestPrettyPrintWithConfigsAndSecrets(t *testing.T) {
+	s := formatServiceInspect(t, formatter.NewServiceFormat("pretty"), time.Now())
+
+	if !strings.Contains(s, "Configs:") {
```
Comment (silvin-lubecki) [inline cli/command/service/inspect_test.go]: You can use gotestyourself Contains assertion here:
```go
assert.Check(t, is.Contains(s, "Configs:"), "Pretty print missing configs")
assert.Check(t, is.Contains(s, "Secrets:"), "Pretty print missing secrets")
```

Comment (thaJeztah) [reply cli/command/service/inspect_test.go]: Wondering if we should use the `.golden` files for this @vdemeester ?

Comment (vdemeester) [reply cli/command/service/inspect_test.go]: Yes, most likely we could use `golden` package for it :angel: 
Can be done in a follow-up though :wink:

**Comment by essamhassan** [comment]
@thaJeztah I got you and I agree with that. I'm working on the changes and will push them asap.

**Comment by thaJeztah** [comment]
Thanks!

**Comment by thaJeztah** [comment]
Looking good:

```
ID:		z80tan5r9j7bfqypz1am1hrgi
Name:		testie
Configs:
 Target:	/foo/config
  Source:	my_config
 Target:	/bar/config
  Source:	my_config
Secrets:
 Target:	/foo/secret
  Source:	my_secret
 Target:	/bar/secret
  Source:	my_secret
Service Mode:	Replicated
 Replicas:	1
Placement:
UpdateConfig:
 Parallelism:	1
 On failure:	pause
 Monitoring Period: 5s
 Max failure ratio: 0
 Update order:      stop-first
RollbackConfig:
 Parallelism:	1
 On failure:	pause
 Monitoring Period: 5s
 Max failure ratio: 0
 Rollback order:    stop-first
ContainerSpec:
 Image:		nginx:alpine@sha256:3a44395131c5a9704417d19ab4c8d6cb104013659f5babb2f1c632e789588196
Mounts:
  Target = /bax
   Source = foobar
   ReadOnly = false
   Type = volume
  Target = /baz
   Source = foobar
   ReadOnly = false
   Type = volume
Resources:
Networks: testnet 
Endpoint Mode:	vip
```

**Comment by thaJeztah** [comment]
So, one thing (bear with me) I'm thinking while looking at the output: should we group "attached" objects, and move `configs` and `secrets` lower (near `Mounts` and `Networks`)? Perhaps `Labels` should be moved as well (not sure about that one though, and that should be separate from this PR); would look something like;

```
ID:		x2l4mdjijqncfeebbq7zmz5tr
Name:		testie
Service Mode:	Replicated
 Replicas:	1
Placement:
UpdateConfig:
 Parallelism:	1
 On failure:	pause
 Monitoring Period: 5s
 Max failure ratio: 0
 Update order:      stop-first
RollbackConfig:
 Parallelism:	1
 On failure:	pause
 Monitoring Period: 5s
 Max failure ratio: 0
 Rollback order:    stop-first
ContainerSpec:
 Image:		nginx:alpine@sha256:3a44395131c5a9704417d19ab4c8d6cb104013659f5babb2f1c632e789588196
Labels:
 bar=baz
 foo=bar
Configs:
 Target:	/bar/config
  Source:	my_config
 Target:	/foo/config
  Source:	my_config
Secrets:
 Target:	/foo/secret
  Source:	my_secret
 Target:	/bar/secret
  Source:	my_secret
Mounts:
  Target = /bax
   Source = foobar
   ReadOnly = false
   Type = volume
  Target = /baz
   Source = foobar
   ReadOnly = false
   Type = volume
Resources:
Networks: testnet 
Endpoint Mode:	vip
```

**Comment by essamhassan** [comment]
Makes sense but I'd say if we are going to follow the same order as in the json output they should be after Mounts

```bash
ID:		x2l4mdjijqncfeebbq7zmz5tr
Name:		testie
Service Mode:	Replicated
 Replicas:	1
Placement:
UpdateConfig:
 Parallelism:	1
 On failure:	pause
 Monitoring Period: 5s
 Max failure ratio: 0
 Update order:      stop-first
RollbackConfig:
 Parallelism:	1
 On failure:	pause
 Monitoring Period: 5s
 Max failure ratio: 0
 Rollback order:    stop-first
ContainerSpec:
 Image:		nginx:alpine@sha256:3a44395131c5a9704417d19ab4c8d6cb104013659f5babb2f1c632e789588196
Labels:
 bar=baz
 foo=bar
Mounts:
  Target = /bax
   Source = foobar
   ReadOnly = false
   Type = volume
  Target = /baz
   Source = foobar
   ReadOnly = false
   Type = volume
Configs:
 Target:	/bar/config
  Source:	my_config
 Target:	/foo/config
  Source:	my_config
Secrets:
 Target:	/foo/secret
  Source:	my_secret
 Target:	/bar/secret
  Source:	my_secret
Resources:
Networks: testnet 
Endpoint Mode:	vip
```

**Comment by thaJeztah** [comment]
Ah! Definitely ok with me to move after mounts

**Pull request merged**